runbotics/runbotics-orchestrator-ui/src/src-app/views/users/UsersRegisterView/UsersRegisterView.tsx

```diff
@@ -1,9 +1,147 @@
-import React, { FunctionComponent } from 'react';
+import React, { FC, useState, useEffect } from 'react';
 
-const UsersRegisterView: FunctionComponent = () => {
+import { useSearchParams } from 'next/navigation';
+import { useRouter } from 'next/router';
+import { useSnackbar } from 'notistack';
+import { useSelector } from 'react-redux';
+import { Role, IUser } from 'runbotics-common';
+
+import useTranslations from '#src-app/hooks/useTranslations';
+import useUserSearch from '#src-app/hooks/useUserSearch';
+import { useDispatch } from '#src-app/store';
+import { usersActions, usersSelector } from '#src-app/store/slices/Users';
+
+import { DefaultPageValue, ROWS_PER_PAGE } from '../UsersBrowseView/UsersBrowseView.utils';
+import UsersRegisterTable from './UsersRegisterTable';
+import { StyledButtonsContainer, StyledButton, DeleteButton, StyledActionsContainer, StyledTextField } from './UsersRegisterView.styles';
+
+interface SelectedRoles { [id: number]: Role };
+interface MapActivatedUserParams {
+    id: number;
+    login: string;
+    roles: Role[];
+    activated: boolean;
+}
+
+const UsersRegisterView: FC = () => {
+    const { enqueueSnackbar } = useSnackbar();
+    const { translate } = useTranslations();
+    const dispatch = useDispatch();
+    const router = useRouter();
+    const searchParams = useSearchParams();
+
+    const currentPage = searchParams.get('page');
+    const pageSizeFromUrl = searchParams.get('pageSize');
+    const [page, setPage] = useState(currentPage ? parseInt(currentPage, 10) : DefaultPageValue.PAGE);
+    const [limit, setLimit] = useState(
+        pageSizeFromUrl && ROWS_PER_PAGE.includes(parseInt(pageSizeFromUrl, 10))
+            ? parseInt(pageSizeFromUrl, 10)
+            : DefaultPageValue.PAGE_SIZE);
+
+    const { notActivated } = useSelector(usersSelector);
+    const { search, handleSearch, refreshSearch } = useUserSearch(false, limit, page);
+
+    const [selectedRoles, setSelectedRoles] = useState<SelectedRoles>({});
+    const [selections, setSelections] = useState<number[]>([]);
+
+    const handleSelectedRolesChange = (id: number, value: Role) =>
+        setSelectedRoles({ ...selectedRoles, [id]: value});
+
+    const handleSelectionChange = (selection) => setSelections(selection);
+
+    const handleAccept = () => {
+        const hasEveryUserSelectedRole = selections.every(checkUserRole);
+        if (hasEveryUserSelectedRole) {
+            const payload = selections.map((userId) => {
+                const role = selectedRoles[userId];
+                return mapActivatedUserRequest(userId, role);
+            });
+
+            handleSubmit(payload);
+        } else {
+            enqueueSnackbar(translate('Users.Register.View.Events.Error.RolesNotSelected'), { variant: 'error' });
+        }
```

Comment:
to minimize indents you can change to 
if (!hasEveryUserSelectedRole) {
enqueueSnackbar(translate('Users.Register.View.Events.Error.RolesNotSelected'), { variant: 'error' });
return;
}
const payload = selections.map((userId) => {
                const role = selectedRoles[userId];
                return mapActivatedUserRequest(userId, role);
            });

            handleSubmit(payload);